File: swap.jl
# Tested on Julia 0.3

macro swap!(x, y)
    quote
        tmp = $(esc(x))
        $(esc(x)) = $(esc(y))
        $(esc(y)) = tmp
    end
end

a = 1
b = 2

@swap!(a, b)

println("a: $a, b: $b")

File: swap.js
// Compile with:
// $ sjs -o output.js swap.js
// Run with:
// $ node output.js

macro swap {
    rule { ($x, $y) } => {
        var tmp = $x;
        $x = $y;
        $y = tmp;
    }
}

var a = 1, b = 2;
swap(a, b);
console.log([a, b]);
